File: frontend/app/types/user_wanted_pages_filter.ts
export type UserWantedPageCriteria = {
    name: string
    siteNames?: string[]
    keywords?: string[]
    titleContains?: string
}

export type UserWantedPagesFilter = {
    id: number
    userId: number
    createdAt: Date
    updatedAt: Date
    deletedAt?: Date
    filterData: UserWantedPageCriteria
}

File: app/internal/model/user_wanted_pages_filter.go
package model

import "github.com/lib/pq"

type UserWantedPagesFilter struct {
	FilterData UserWantedPageCriteria `json:"filterData" gorm:"embedded"`
	User       User                    `json:"-" gorm:"constraint:OnUpdate:CASCADE,OnDelete:RESTRICT;"`
	Model
	UserID uint `json:"userId" gorm:"not null;column:user_id;"`
}

type UserWantedPageCriteria struct {
	Name          string         `json:"name" gorm:"column:name;not null;default:''"`
	SiteNames     pq.StringArray `json:"siteNames,omitempty" gorm:"column:site_names;type:text[]"`
	Keywords      pq.StringArray `json:"keywords,omitempty" gorm:"column:keywords;type:text[]"`
	TitleContains string         `json:"titleContains,omitempty" gorm:"column:title_contains"`
}
